src/edge_cov.rs

```diff
@@ -0,0 +1,107 @@
+use alloy_primitives::{Address, U256};
+use revm::{
+    interpreter::{opcode::OpCode, Interpreter},
+    Database, EvmContext, Inspector,
+};
+
+use std::hash::{Hash, Hasher};
+
+/// A hit count that never goes to zero e.g. the back edge of a loop that iterates 256 times will be
+/// one instead of zero.
+// see https://github.com/AFLplusplus/AFLplusplus/blob/5777ceaf23f48ae4ceae60e4f3a79263802633c6/instrumentation/afl-llvm-pass.so.cc#L810-L829
+#[derive(Clone, Copy, Debug, Default)]
+pub struct NeverZeroHitCount(pub u8);
+
+impl std::ops::Add for NeverZeroHitCount {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self {
+        Self(self.0.checked_add(rhs.0).unwrap_or(1))
+    }
+}
+
+impl std::ops::AddAssign for NeverZeroHitCount {
+    fn add_assign(&mut self, rhs: Self) {
+        *self = *self + rhs;
+    }
+}
+
+const MAX_EDGE_COUNT: usize = 65536;
+
+/// An Inspector that tracks edge coverage.
+#[derive(Clone, Debug, Default)]
+pub struct EdgeCovInspector {
+    /// Map of total gas used per opcode.
+    pub hitcount: Vec<NeverZeroHitCount>,
+}
+
+impl EdgeCovInspector {
+    /// Create a new EdgeCovInspector.
+    pub fn new() -> Self {
+        Self { hitcount: vec![NeverZeroHitCount(0); MAX_EDGE_COUNT] }
+    }
+
+    /// Reset the hitcount to zero.
+    pub fn reset(&mut self) {
+        self.hitcount.fill(NeverZeroHitCount(0));
+    }
+}
+
+fn edge_hash(address: Address, pc: usize, jump_dest: U256) -> u32 {
+    // TODO faster hash https://h0mbre.github.io/Lucid_Snapshots_Coverage/
+    let mut hasher = std::collections::hash_map::DefaultHasher::new();
```

Comment:
we could also use alloy_primitives::map::DefaultHasher here

Reply:
should be DefaultHashBuilder::default().build_hasher()

Reply:
I think @DaniPopes is suggesting

`alloy_primitives::map::DefaultHashBuilder::default().build_hasher()`

the default hasher is chosen at compile time, depending on the features

Reply:
I tried this hasher and it doesn't produce the same edge hash for the same inputs. It seems there is a per hasher seed. Ig maybe there's some way to make this global and have the inspector maintain the state?

Reply:
ah right of course, you want to store the hash builder in the struct and use that to .build_hasher()